aws/resources/ec2_network_acl.go

```diff
@@ -0,0 +1,251 @@
+		// check first seen tag
+		firstSeenTime, err := nacl.getFirstSeenTag(*networkAcl)
+		if err != nil {
+			logging.Errorf(
+				"Unable to retrieve tags for network acl: %s, with error: %s", aws.StringValue(networkAcl.NetworkAclId), err)
+			continue
+		}
+
+		// if the first seen tag is not there, then create one
+		if firstSeenTime == nil {
+			now := time.Now().UTC()
+			firstSeenTime = &now
+			if err := nacl.setFirstSeenTag(*networkAcl, time.Now().UTC()); err != nil {
+				logging.Errorf(
+					"Unable to apply first seen tag network acl: %s, with error: %s", aws.StringValue(networkAcl.NetworkAclId), err)
+				continue
+			}
+		}
+
+		if shouldIncludeNetworkACL(networkAcl, firstSeenTime, configObj) {
+			identifiers = append(identifiers, networkAcl.NetworkAclId)
+		}
+	}
+
+	// Check and verify the list of allowed nuke actions
+	nacl.VerifyNukablePermissions(identifiers, func(id *string) error {
+		_, err := nacl.Client.DeleteNetworkAcl(&ec2.DeleteNetworkAclInput{
+			NetworkAclId: id,
+			DryRun:       aws.Bool(true),
+		})
+		return err
+	})
+
+	return identifiers, nil
+}
+
+func (nacl *NetworkACL) nuke(id *string) error {
+
+	// nuke attached subnets
+	if err := nacl.nukeAssociatedSubnets(id); err != nil {
+		return errors.WithStackTrace(err)
+	}
+
+	// nuking the network ACL
+	if err := nacl.nukeNetworkAcl(id); err != nil {
+		return errors.WithStackTrace(err)
+	}
+
+	return nil
+}
+
+// nukeAssociatedSubnets dissociates the specified network ACL from associated subnets and replaces the associations with the default network ACL in the same VPC.
+// Important : You can't delete the ACL if it's associated with any subnets. You can't delete the default network ACL.
+//
+// Thus, to remove the association, it requires another network ACL ID. Here, we check the default network ACL of the VPC to which the current network ACL is attached,
+// and then associate that network ACL with the association.
+func (nacl *NetworkACL) nukeAssociatedSubnets(id *string) error {
+	resp, err := nacl.Client.DescribeNetworkAcls(&ec2.DescribeNetworkAclsInput{
+		NetworkAclIds: []*string{id},
+	})
+	if err != nil {
+		logging.Debugf("[Network ACL] Failed to describe network ACL: %s", err)
+		return err
+	}
+
+	if len(resp.NetworkAcls) <= 0 {
+		logging.Debugf("[Network ACL] Nothing found: %s", aws.StringValue(id))
+		return nil
+	}
+
+	var (
+		networkAcl = resp.NetworkAcls[0]
+		vpcID      = networkAcl.VpcId // get the vpc of this nacl
+	)
+
+	// Get the default nacl association id
+	networkACLs, err := nacl.Client.DescribeNetworkAcls(
+		&ec2.DescribeNetworkAclsInput{
+			Filters: []*ec2.Filter{
+				{
+					Name:   aws.String("vpc-id"),
+					Values: []*string{vpcID},
+				}, {
+					Name: aws.String("default"),
+					Values: []*string{
+						aws.String("true"),
+					},
+				},
+			},
+		},
+	)
+	if err != nil {
+		logging.Debugf("[Network ACL] Failed to describe network ACL: %s", err)
+		return err
+	}
+
+	if len(networkACLs.NetworkAcls) <= 0 {
```

Comment:
I suspect the slice length can't be bellow 0

Reply:
Will remove.